src/components/FormDatePicker.vue

```diff
@@ -32,157 +26,86 @@
+      valueZone: {
+        type: String,
+        default() {
+          if (typeof ProcessMaker !== 'undefined' && ProcessMaker.user && ProcessMaker.user.app_timezone) {
+            return ProcessMaker.user.app_timezone;
+          }
+
+          return 'UTC';
         }
-        return controlClass;
       },
-      dtMin() {
-        if (this.minDatetime) {
-          return this.minDatetime;
+      zone: {
+        type: String,
+        default() {
+          if (typeof ProcessMaker !== 'undefined' && ProcessMaker.user && ProcessMaker.user.timezone) {
+            return ProcessMaker.user.timezone;
+          }
+
+          return 'local';
         }
-        return "1";
       },
-      dtMax() {
-        if (this.maxDatetime) {
-          return this.maxDatetime;
+      weekStart: { type: Number, default: 7 },
+      format: {
+        type: [String, Object],
+        default() {
+          if (typeof ProcessMaker !== 'undefined' && ProcessMaker.user && ProcessMaker.user.calendar_format) {
+            return ProcessMaker.user.calendar_format;
+          }
+
+          return {
+            year: 'numeric',
+            month: 'numeric',
+            day: 'numeric',
+            hour: 'numeric',
+            minute: '2-digit',
+          };
         }
-        return "1";
       },
-      dtFormat() {
-        //If the global variable is set by default.
-        if (typeof ProcessMaker !== "undefined" && ProcessMaker.user && ProcessMaker.user.calendar_format) {
-          return ProcessMaker.user.calendar_format;
+      phrases: {
+        type: [String, Object],
+        default() {
+          return { ok: 'Continue', cancel: 'Exit' };
         }
-        if (this.format && typeof this.format === 'string') {
-          return this.format;
-        }
-        let format = {};
-        format.year = 'numeric';
-        format.month = 'numeric';
-        format.day = 'numeric';
-        format.hour = 'numeric';
-        format.minute = '2-digit';
-        if (this.format && typeof this.format === 'object') {
-          format.year = this.format.year ? this.format.year : format.year;
-          format.month = this.format.month ? this.format.month : format.month;
-          format.day = this.format.day ? this.format.day : format.day;
-          format.hour = this.format.hour ? this.format.hour : format.hour;
-          format.minute = this.format.minute ? this.format.minute : format.minute;
-
-          if (this.format.timeZoneName) {
-            format.timeZoneName = this.format.timeZoneName;
-          }
-        }
-        return format;
       },
-      dtPhrases() {
-        let phrases = {};
-        phrases.ok = 'Continue';
-        phrases.cancel = 'Exit';
-        if (this.phrases && typeof this.phrases === 'string') {
+      auto: { type: Boolean, default: true },
+    },
+    computed: {
+      parsedPhrases() {
+        let phrases;
+
+        if (typeof this.phrases === 'string') {
```

Comment:
Will a string actually ever be passed in? I think having this here adds unneeded complexity.